### PR TITLE
Remove unnecessary dotcom checks

### PR DESCRIPTION
**Pull request by eseliger**
These checks for dotcom mode are no longer necessary, so let's remove the number of code paths.

Test plan:

Test suites are still passing, also code review.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/org.go [hunk 1]
```diff
@@ -443,12 +434,6 @@ func (r *schemaResolver) AddUserToOrganization(ctx context.Context, args *struct
 		return nil, err
 	}
 
-	// 🚨 SECURITY: Do not allow direct add on Cloud unless the site admin is a member of the org
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/org.go:446]: @evict this seems ok, does it?

We used to only allow admins that are _also_ members of an org add a member to said org.

That prevented admins to gain access to an orgs private code in v1 cloud, which no longer exists.

Comment (evict) [reply cmd/frontend/graphqlbackend/org.go:446]: Makes sense!

**Review by evict — APPROVED** [review]
🧹

**Review by unknwon — COMMENTED** [review]
Thank you 😁

**Pull request merged**